logs.go

```diff
@@ -383,6 +383,39 @@ func (w *windowWithLogs) Close() {
 	}
 }
 
+// Controller logs.
+type controllerWithLogs struct {
+	Controller
+}
+
+func (c *controllerWithLogs) Close() {
+	WhenDebug(func() {
+		Logf("controller %s is closing", c.ID())
+	})
+
+	c.Controller.Close()
+	if c.Err() != nil {
+		Logf("controller %s failed to close: %s",
+			c.ID(),
+			c.Err(),
+		)
+	}
+}
+
+func (d *driverWithLogs) NewController(c ControllerConfig) Controller {
```

Comment:
This should be moved with the other functions related to the `driverWithLogs`. Please keep the same order as where you declared it in driver.

Reply:
I see, I swapped `NewController` and `Close`. Is that all you meant? Or do you want me to move the overall placement?

Reply:
Oh! I see it now ha

Reply:
Driver with logs is a different struct than controller with logs

Reply:
yeah that was a subtle difference that went over my head 😄